dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -0,0 +1,516 @@
+        }
+        else
+        {
+            GetClassResponse? existing = JsonSerializer.Deserialize<GetClassResponse>(responseContent, s_jsonSerializerOptions);
+            if (existing != null && existing.Description != ToWeaviateFriendlyClassDescription(collectionName))
+            {
+                // ReSharper disable once CommentTypo
+                // Check that we don't have an accidental conflict.
+                // For example a collectionName of '__this_collection' and 'this_collection' are
+                // both transformed to the class name of SKthiscollection - even though the external
+                // system could consider them as unique collection names.
+                if (existing.Description != ToWeaviateFriendlyClassDescription(collectionName))
+                {
+                    throw new WeaviateMemoryException(WeaviateMemoryException.ErrorCodes.CollectionNameConflict, $"Unable to verify existing collection: {collectionName} with class name: {className}");
+                }
+            }
+        }
+
+        return exists;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> GetCollectionsAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        this._log.LogTrace("Listing collections");
+
+        using HttpRequestMessage request = GetSchemaRequest.Create().Build();
+        (HttpResponseMessage _, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        GetSchemaResponse? getSchemaResponse = JsonSerializer.Deserialize<GetSchemaResponse>(responseContent, s_jsonSerializerOptions);
+
+        foreach (GetClassResponse? @class in getSchemaResponse!.Classes!
+                     .Where(@class => IsPossibleSemanticKernelCollection(@class.Class!)))
+        {
+            yield return @class.Class!;
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task DeleteCollectionAsync(string collectionName, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+
+        if (await this.DoesCollectionExistAsync(collectionName, cancellationToken).ConfigureAwait(false))
+        {
+            string className = ToWeaviateFriendlyClassName(collectionName);
+            this._log.LogTrace("Deleting collection: {0}, with class name: {1}", collectionName, className);
+
+            using HttpRequestMessage request = DeleteSchemaRequest.Create(className).Build();
+            (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException e)
+            {
+                this._log.LogError(e, "Collection deletion failed: {0}, {1}", e.Message, responseContent);
+                throw;
```

Comment:
In some methods in case of `HttpRequestException` we throw `WeaviateMemoryException` with error code, but here for example we re-throw `HttpRequestException`. It would be nice if we align all method responses/exceptions to the same format/type.

Reply:
Sure.